src/AppRoutes.js

```diff
@@ -7,43 +7,39 @@ import PropTypes from 'prop-types';
 import { TabLayout } from './layouts';
 import { SignIn as SignInView, AccessDenied as AccessDeniedView } from './views';
 import Workspaces from './views/Workspaces';
-import { useWorkspace } from './state/hooks/WorkspaceHooks';
+import { useWorkspacesList } from './state/hooks/WorkspaceHooks';
 
 const AppRoutes = (props) => {
   const { authenticated, authorized, tabs } = props;
   const location = useLocation();
-  const previousUrl = sessionStorage.getItem('previousURL');
-  const currentWorkspace = useWorkspace();
+  const providedUrl = sessionStorage.getItem('providedUrl');
+  const providedUrlBeforeSignIn = sessionStorage.getItem('providedUrlBeforeSignIn');
+  const workspacesList = useWorkspacesList();
+
   return (
     <Routes>
-      <Route path="/" element={<Navigate to={previousUrl || '/workspaces'} replace />} />
+      <Route path="/" element={<Navigate to={providedUrlBeforeSignIn || providedUrl || '/workspaces'} replace />} />
       <Route
         path="/workspaces"
         element={
           !authenticated ? (
             <Navigate to={'/sign-in'} state={{ from: location.pathname }} />
           ) : !authorized ? (
-            <Navigate to={'/accessDenied'} />
-          ) : currentWorkspace?.data ? (
-            <Navigate to={`/${currentWorkspace.data.id}/scenario`} />
-          ) : (
+            <Navigate to="/accessDenied" replace />
+          ) : workspacesList?.length > 1 ? (
```

Comment:
Checking `workspacesList?.length > 1` to redirect to the "workspaces" view does not handle the case where the number of workspaces is 0 (we want to show a placeholder in the "workspaces" view in this case).
Maybe we could check `workspacesList?.length !== 1` instead ?